DiscordChatExporter.Core/Exporting/Filtering/HasMessageFilter.cs

```diff
@@ -0,0 +1,28 @@
+﻿using DiscordChatExporter.Core.Discord.Data;
+using System.Linq;
+
+namespace DiscordChatExporter.Core.Exporting.Filtering
+{
+    public class HasMessageFilter : MessageFilter
+    {
+        private readonly string _value;
+
+        public HasMessageFilter(string value)
+        {
+            ///TODO: should we validate input here?
+            _value = value;
+        }
+
+        public override bool Filter(Message message) =>
+            _value switch
+            {
+                "link" => throw new System.NotImplementedException(), //how do we do this?
```

Comment:
I'm not actually 100% sure how Discord checks for links. I'd assume just some sort of regex on the message content?

Reply:
I think we identify links currently when parsing for markdown, the links are parsed into instances of `LinkNode`.

Reply:
Could we get away with just throwing a regex in here? `https?:\/\/\S\S` seems to match Discord's behaviour well enough (except for edge cases like "http://.." where the link body consists of only two of certain punctuation characters).

Reply:
You can take this regex here (it's a bit more elaborate): https://github.com/Tyrrrz/DiscordChatExporter/blob/95cd6cb50cc1498a3512818c50f9990be0d01421/DiscordChatExporter.Core/Markdown/MarkdownParser.cs#L189-L192

Later we might figure out a way to unify it, but for now we can just copy-paste